common/db/db.go

```diff
@@ -491,3 +491,88 @@ func UploadFileGridFS(file io.Reader, fileName string, force bool) error {
 		return nil
 	}
 }
+
+func GetGlobalStreamSettings() (models.StreamSettings, error) {
+	var globalSettings models.GlobalSettings
+	var streamSettings models.StreamSettings
+
+	coll := mongoClient.Database("gads").Collection("global_settings")
+	filter := bson.D{{Key: "type", Value: "stream-settings"}}
+
+	err := coll.FindOne(mongoClientCtx, filter).Decode(&globalSettings)
+	if err == mongo.ErrNoDocuments {
+		streamSettings = models.StreamSettings{
+			TargetFPS:            15,
+			JpegQuality:          75,
+			ScalingFactorAndroid: 50,
+			ScalingFactoriOS:     50,
+		}
+
+		err = UpdateGlobalStreamSettings(streamSettings)
+		if err != nil {
+			return streamSettings, err
+		}
+	} else if err != nil {
+		return streamSettings, err
+	} else {
+		settingsBytes, err := bson.Marshal(globalSettings.Settings)
+		if err != nil {
+			return streamSettings, fmt.Errorf("failed to marshal settings: %v", err)
+		}
+
+		err = bson.Unmarshal(settingsBytes, &streamSettings)
+		if err != nil {
+			return streamSettings, fmt.Errorf("failed to unmarshal settings: %v", err)
+		}
+	}
+
+	return streamSettings, nil
+}
+
+func UpdateGlobalStreamSettings(settings models.StreamSettings) error {
+	globalSettings := models.GlobalSettings{
+		Type:        "stream-settings",
+		Settings:    settings,
+		LastUpdated: time.Now(),
+	}
+
+	update := bson.M{
+		"$set": globalSettings,
+	}
+
+	coll := mongoClient.Database("gads").Collection("global_settings")
+	filter := bson.D{{Key: "type", Value: "stream-settings"}}
+	opts := options.Update().SetUpsert(true)
+
+	_, err := coll.UpdateOne(mongoClientCtx, filter, update, opts)
+	return err
+}
+
+func GetDeviceStreamSettings(udid string) (models.DeviceStreamSettings, error) {
+	var deviceStreamSettings models.DeviceStreamSettings
+
+	coll := mongoClient.Database("gads").Collection("device_stream_settings")
+	filter := bson.D{{Key: "udid", Value: udid}}
+
+	err := coll.FindOne(mongoClientCtx, filter).Decode(&deviceStreamSettings)
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			return deviceStreamSettings, nil // No settings found, return empty struct
```

Comment:
Why don't we just return the error in both cases? Then we can avoid the unnecessary check for device UDID in `provider/devices/common.go` and just fallback to the default stream settings

Reply:
Ok, it makes sense. I will work on it